x11/xauth_test.go

```diff
@@ -0,0 +1,76 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package x11_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/x11"
+)
+
+// Hook up check.v1 into the "go test" runner
+func Test(t *testing.T) { TestingT(t) }
+
+type XauthTestSuite struct {}
+
+var _ = Suite(&XauthTestSuite{})
+
+func (s *XauthTestSuite) TestXauthFileNotAvailable(c *C) {
+	n, err := x11.ValidateXauthority("/does/not/exist")
+	c.Assert(n, Equals, 0)
+	c.Assert(err, Not(IsNil))
+}
+
+func (s *XauthTestSuite) TestXauthFileExistsButIsEmpty(c *C) {
+	f, err := ioutil.TempFile("", "xauth")
```

Comment:
(nitpick^99) - you could use `tempdir := c.MkDir(); ioutil.WriteFile(filepath.Join(tempfile, "xauth"), nil, 0600)` here which is a whopping **one** line shorter ;) so really not worth it. Mostly wanted to show `c.MkDir()` as its nice and the gocheck code will do the cleanup for us of that temp dir (you could create a tempdir in in `SetupTest()` and use use that for the tempfiles without having to worry about cleanup. But I'm really into nitpick land now :)